src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -311,71 +378,54 @@ void ZoneWindow::InitializeId(PCWSTR deviceId, PCWSTR virtualDesktopId) noexcept
 
 void ZoneWindow::LoadSettings() noexcept
 {
-    wchar_t activeZoneSetId[256];
-    RegistryHelpers::GetString(m_uniqueId, L"ActiveZoneSetId", activeZoneSetId, sizeof(activeZoneSetId));
-    CLSIDFromString(activeZoneSetId, &m_activeZoneSetId);
+    auto& deviceInfoMap = JSONHelpers::FancyZonesDataInstance().GetDeviceInfoMap();
+    if (!deviceInfoMap.contains(m_uniqueId))
+    {
+        JSONHelpers::FancyZonesDataInstance().MigrateDeviceInfoFromRegistry(m_uniqueId);
+    }
+  
+    JSONHelpers::FancyZonesDataInstance().GetDeviceInfoFromTmpFile(m_uniqueId, m_activeZoneSetPath);
+    const WCHAR* activeZoneSetStr = deviceInfoMap[m_uniqueId].activeZoneSet.uuid.c_str();
```

Comment:
If app is newly installed and there is no data in registry and temp file, there will be created new empty device info. Is it ok?

Reply:
It's initialized with default values (1-zone grid zoneset). Only thing missing is to generate valid uuid, so far it's empty. I have it in my todo list